plugins/catalog-import/src/api/AzureRepoApiClient.ts

```diff
@@ -0,0 +1,269 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export interface AzurePrOptions {
+  tenantUrl: string;
+  title: string;
+  project: string;
+  branchName: string;
+  repository: string;
+  token: string;
+  fileContent: string;
+  fileName: string;
+  description: string;
+}
+
+export interface CreateAzurePr {
+  sourceRefName: string;
+  targetRefName: string;
+  title: string;
+  description: string;
+}
+
+export interface AzureRepo {
+  id: string;
+  name: string;
+  defaultBranch: string;
+}
+
+export interface AzureRef {
+  name: string;
+  objectId: string;
+}
+
+interface RefQueryResult {
+  value: AzureRef[];
+}
+
+export interface AzureCommit {
+  comment: string;
+
+  changes: {
+    changeType: string;
+    item: {
+      path: string;
+    };
+    newContent: {
+      content: string;
+      contentType: 'rawtext';
+    };
+  }[];
+}
+
+export interface AzureRefUpdate {
+  repositoryId: string;
+  name: string;
+  oldObjectId: string;
+  newObjectId: string;
+}
+
+export interface AzurePushResult {
+  refUpdates: AzureRefUpdate[];
+}
+
+export interface AzurePush {
+  refUpdates: {
+    name: string;
+    oldObjectId: string;
+  }[];
+  commits: AzureCommit[];
+}
+
+export interface AzurePrResult {
+  pullRequestId: string;
+  repository: {
+    name: string;
+    webUrl: string;
+  };
+}
+
+const apiVersions = {
```

Comment:
We currently support `6.0` of the Azure DevOps API - https://github.com/backstage/backstage/blob/d26e5a5d45ee34d3633df8f878477ccd22d947ae/packages/integration/src/azure/AzureUrl.ts#L138 - we should target that here as well.

Reply:
Sorry, I should have been more clear. Your changes should be updated to support version `6.0` of the Azure DevOps API. There are Backstage Adopters using this version and will expect this feature to work for them as every other area of Backstage that supports Azure DevOps works with their version of the API.

Are you sure the version `6.0` for these end points doesn't work? It's been a long time but I have high confidence that this works as I've done it in PowerShell before just using the REST endpoints. 🤔

Reply:
I have tested the endpoints with Versions prior to 7 and not all  appeared to work, I'm afraid :( . So  I narrowed them down the common denominator. I can look a bit more, maybe I will find alternatives  with V6
Will come back with updates and maybe some references

Reply:
Ok. I was wrong. It appears to be working perfectly fine with V6. I will make the changes and update the PR